index.js

```diff
@@ -2,44 +2,53 @@
 
 const isPrimitive = value => value === null || (typeof value !== 'object' && typeof value !== 'function');
 
+const concatPath = (path, property) => {
+	if (property && property.toString) {
+		if (path) {
+			path += '.';
+		}
+		path += property.toString();
+	}
+	return path;
+};
+
 const proxyTarget = Symbol('ProxyTarget');
 
 module.exports = (object, onChange) => {
 	let inApply = false;
 	let changed = false;
 	const propCache = new WeakMap();
+	const pathCache = new WeakMap();
 
-	const handleChange = () => {
+	const handleChange = (path, property, previous, value) => {
 		if (!inApply) {
-			onChange();
+			onChange.call(proxy, concatPath(path, property), previous, value);
```

Comment:
It's a small thing, but I suspect there are going to be use-cases where the application only cares about the new value. In other change-notification APIs I've seen, it's somewhat common to provide the new value before the previous value, so callbacks can simple drop the last parameter if they don't care about the previous value.

Consider this signature instead: `onChange(value, previous, path)`.

It's pretty unlikely that an application would care about the previous value but *not* the new value. I'm less sure that moving the path to the end of the argument list, but given that the current version doesn't provide this information at all, this "feels" right intuitively.

Like I said, this is a small thing, so take this suggestion under advisement :)

Reply:
Putting the value before the previous makes sense to me, I'll make that change. But does the value make sense without the path on a deeply nested object? How would you feel about `onChange(path, value, previous)`?

Reply:
Yeah, that was another option I was going to suggest :)